test/domains.jl

```diff
@@ -116,3 +116,63 @@ end
               parent(coordinates(column))
     end
 end
+
+TestFloatTypes = (Float32, Float64)
+@testset "RootDomain" begin
+    for FT in TestFloatTypes
+        d_z = 1.0
+        roots = [1.0, 2.0, 3.0]
+        stem = Int64(5)
+        leaves = Int64(4)
+        comp_heights = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0]
+        comp_labels =
+            [:stem, :stem, :stem, :stem, :stem, :leaf, :leaf, :leaf, :leaf]
+        test_tuple = (root = 1, stem = 2, leaf = 3)
+        @test test_tuple[:root] == 1
+
+        testing = RootDomain(roots, stem, leaves, d_z)
```

Comment:
if you remove your third constructor, could still test the default constructor here...